Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -113,6 +130,27 @@ extension SwiftLanguageServer {
       guard let self = self else { return }
       let notification = SKResponse(notification, sourcekitd: self.sourcekitd)
 
+      if notification.value?[self.keys.notification] == self.values.notification_sema_enabled {
+        self.state = .connected
+      }
+
+      if self.state == .connectionInterrupted {
+        // If we get a notification while we are restoring the connection, it means that the server has restarted.
+        // We still need to wait for semantic functionality to come back up.
+        self.state = .semanticFunctionalityDisabled
+
+        // Ask our parent to re-open all of our documents.
+        self.reopenDocuments()
+      }
+
+      if let error = notification.error, error.code == .connectionInterrupted {
+        self.state = .connectionInterrupted
+
+        // We don't have any open documents anymore after sourcekitd crashed.
```

Comment:
I don’t think so. Even though `sourcekitd` crashed, the build settings should still be correct, so we should be able to keep them around. Do you agree?

Reply:
Yeah that seems reasonable although if the re-open fails you might never wipe them

Reply:
Well, even if the re-open fails, the build settings are correct. They are just no longer used. I don’t see a reason to wipe them ever.

Reply:
Yeah, it shouldn't be a big deal but in theory if it crashes with a lot of documents open it will save all of the build settings. But I don't think it's a major concern

Reply:
Who will save the build settings?

Also: What’s the problem with saving the build settings anyway. They are still correct.

Reply:
Well `SwiftLanguageServer` does, imagine a case:
- SourceKitServer has the file open
- SwiftLanguageServer has the file open and saved build settings
- SwiftLanguageServer crashes, reopen fails
- Potential build system changes?

But yeah I don't think this is really an issue

Reply:
I assume by `SwiftLanguageServer`, you mean `sourcekitd` (the `SwiftLanguageServer` never writes anything to disk).

And if `sourcekitd` fails to re-open the file, well, there’s no way we would be able to tell it about the new build settings anyway. Keep in mind that `buildSettingsByFile` keeps being updated even if `sourcekitd` has crashed.

Reply:
Ah I see, for some reason I thought the `buildSettingsByFile` wouldn't be updated anymore